src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep2.tsx

```diff
@@ -0,0 +1,575 @@
+    setAccessorProfiles(filtered)
+
+    // Update form submission request object
+    const newAccessorChanges:AccessorChange[] = filtered.map(item => {
+      return {
+        userId: item.ownerId,
+        type: AccessType.GAIN_ACCESS
+      }
+    })
+    setFormSubmitRequestObject(prevState => {
+      return Object.assign({}, prevState, {
+        accessorChanges: newAccessorChanges
+      })
+    })
+  }
+
+  const onClearAttachment = (fid:string) => {
+    // Update the view
+    const filtered:DataAccessDoc[] = attachments.filter(item => item.fileHandleId !== fid)
+    setAttachments(filtered)
+
+    // Update form submission request object
+    const newAttachments:string[] = filtered.map(item => item.fileHandleId)
+    setFormSubmitRequestObject(prevState => {
+      return Object.assign({}, prevState, {
+        attachments: newAttachments
+      })
+    })
+  }
+
+  const uploadCallback = (data: UploadCallbackResp) => {
+    if (data.context === "attachments") {
+      const docs = formSubmitRequestObject?.attachments ? formSubmitRequestObject?.attachments : []
+      docs?.push(data.resp.fileHandleId)
+      setFormSubmitRequestObject(prevState => {
+        return Object.assign({}, prevState, {"attachments": docs})
+      })
+      // Update the view
+      setAttachments(prev => [...prev, {
+        fileName: data.resp.fileName,
+        fileHandleId: data.resp.fileHandleId
+      }])
+    } else {
+      setFormSubmitRequestObject(prevState => {
+        return Object.assign({}, prevState, {[data.context]: data.resp.fileHandleId})
+      })
+      // Update the view
+      if (data.context === "ducFileHandleId") {
+        setDUC(prevState => {
+          return Object.assign({}, prevState, {
+            fileName: data.resp.fileName,
+            fileHandleId: data.resp.fileHandleId
+          })
+        })
+      }
+      if (data.context === "irbFileHandleId") {
+        setIRB(prevState => {
+          return Object.assign({}, prevState, {
+            fileName: data.resp.fileName,
+            fileHandleId: data.resp.fileHandleId
+          })
+        })
+      }
+    }
+  }
+
+  // User search input event handler
+  const onSelectUserCallback = (selected:UserProfile) => {
+    setAccessorProfiles(prev => [...prev, {
+      ownerId: selected.ownerId,
+      firstName: selected.firstName,
+      lastName: selected.lastName,
+      userName: selected.userName
+    }])
+    const selectedAccessor:AccessorChange = {
+      userId: selected.ownerId,
+      type: AccessType.GAIN_ACCESS
+    }
+    const accessorsArr = formSubmitRequestObject?.accessorChanges || []
+    accessorsArr.push(selectedAccessor)
+    setFormSubmitRequestObject(prevState => {
+      return Object.assign({}, prevState, {
+        accessorChanges: accessorsArr
+      })
+    })
+  }
+
+  return (<>
+    <Form className={"access-request-form2"} onSubmit={e => e.preventDefault()}>
```

Comment:
I wonder if this really needs to be a Form (I assume it's for css only).  If you change this to a div (with the "access-request-form2" css class), does it still just work?

Reply:
I'd like to utilize the bootstrap form elements and the library does provide all the CSS for you so it's convenient. But I also prefer to keep the form tag because there are other form elements like buttons and input fields on the same page so I thought it's semantically meaningful to enclose them all within a form tag.